### ISSUE TITLE
Centralize configuration options in the cosmology structure

### ISSUE THREAD

**Issue by EiffL** [DifferentiableUniverseInitiative/jax_cosmo#42]
For now, we have various configuration options distributed everywhere in the code, the idea being
that with a functional API, you can just specify the parameter to the function doing the computation. The problem is that some quantities are being used in multiple places internally by other functions, so we need to make sure that the same options are being used consistently, storing these options in the cosmology seems to make the most sense.

For instance, the choice of transfer function or non-linear prescription currently requires us to carry a specfic function in all functions that need to compute a power spectrum, but this could simply be a flag in the cosmology object. So, let's do that.

**Comment by santiagocasas** [comment]
Would this still allow to keep the **cool** `F = -hessian(likelihood)(cosmo)`. Probably it depends how the code unpacks that cosmo object, no?

**Comment by EiffL** [comment]
Yep that would still work, the cosmo object can store non-differentiable flags that can be used to store this kind of configuration info.
